go-controller/pkg/util/multi_network.go

```diff
@@ -0,0 +1,321 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	ovncnitypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+
+	kapi "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+var ErrorAttachDefNotOvnManaged = errors.New("net-attach-def not managed by OVN")
+var DefaultNetworkNetNameInfo *NetNameInfo
+
+// NetInfo is interface which holds network name information
+// for default network, this is set to nil
+type NetInfo interface {
+	GetNetworkName() string
+	IsSecondary() bool
+	GetPrefix() string
+	AddNAD(nadName string)
+	DeleteNAD(nadName string)
+	HasNAD(nadName string) bool
+}
+
+// NetNameInfo holds the network name information for secondary network if non-nil
+type NetNameInfo struct {
+	// network name
+	netName string
+	// all net-attach-def NAD names for this network, used to determine if a pod needs
+	// to be plumbed for this network
+	nadNames *sync.Map
+}
+
+// GetNetworkName returns the network name
+func (nInfo *NetNameInfo) GetNetworkName() string {
+	if nInfo == nil {
```

Comment:
Oh I see we still need it because the default implementation is basically this
```
var DefaultNetworkNetNameInfo *NetNameInfo
```
Ok, this is a nit but I would actually have a default implementation of NetInfo. Then we don't have to worry about DefaultNetworkNetNameInfo being overwritten and also save allocating the unused sync.Map, etc...

Reply:
ok, will have a default implementation of NetInfo.